pkg/fqdn/dnspoller.go

```diff
@@ -60,6 +60,10 @@ type DNSPoller struct {
 	// config is a copy from when this instance was initialized.
 	// It is read-only once set
 	config Config
+
+	// DNSHistory is the collection of still-valid DNS responses intercepted
+	// for the poller
+	DNSHistory *DNSCache
```

Comment:
Could you also leave a comment here saying this is not protected by the mutex. Perhaps also move it above the mutex since our convention is that everything below is protected.